autopush/web/registration.py

```diff
@@ -307,6 +336,15 @@ def _return_endpoint(self, endpoint_data, new_uaid, router=None):
                        client_info=self._client_info)
         self.finish()
 
+    def _list_channels(self):
+        _, channels = self.ap_settings.message.all_channels(self.uaid)
+        dashed = [str(uuid.UUID(x)) for x in channels]
+        self.write(json.dumps(
+            {"uaid": self.uaid.hex,
```

Comment:
Not necessarily. For some "state challenged" remote calls, it's not always easy to associate a response back to it's request. Having a shared ID between the two can simplify things. In this case, the UAID is required for the URL and the response, plus it's how the client organizes.